staging/src/k8s.io/kubectl/pkg/explain/v2/templates/plaintext.tmpl

```diff
@@ -0,0 +1,50 @@
+{{- $prefix := (ternary "/api" (join "" "/apis/" $.GVR.Group) (empty $.GVR.Group)) -}}
+
+{{- /* Search both cluster-scoped and namespaced-scoped paths for the GVR to find its GVK */}}
+{{- $clusterScopedSearchPath :=  (replace "." "\\." (join "/" $prefix $.GVR.Version $.GVR.Resource)) }}
+{{- $namespaceScopedSearchPath :=  (replace "." "\\." (join "/" $prefix $.GVR.Version "namespaces" "\\{namespace\\}" $.GVR.Resource)) -}}
+
+{{- $key := join "" "{" (join "." "$" "paths" $clusterScopedSearchPath "get" "x-kubernetes-group-version-kind" ) "}" }}
+{{- $key2 := join "" "{" (join "." "$" "paths" $namespaceScopedSearchPath "get" "x-kubernetes-group-version-kind") "}" -}}
+
+{{- $gvk1 := first (first (lookup $key $.Document)) }}
+{{- $gvk2 := first (first (lookup $key2 $.Document)) }}
+{{- $gvk := coalesce $gvk1 $gvk2 }}
+{{- if $gvk.group -}}
+GROUP:      {{ $gvk.group }}
+{{- end }}
+KIND:       {{ $gvk.kind }}
+VERSION:    {{ $gvk.version }}
+
+{{- /* Find definition with this GVK by filtering out the components/schema with the given x-kubernetes-group-version-kind */}}
+{{- $definitionPath := printf "{$[?(@.x-kubernetes-group-version-kind[0].group=='%v')]}" $gvk.group }}
+{{- $definitionPath2 := printf "{$[?(@.x-kubernetes-group-version-kind[0].version=='%v')]}" $gvk.version }}
+{{- $definitionPath3 := printf "{$[?(@.x-kubernetes-group-version-kind[0].kind=='%v')]}" $gvk.kind }}
+{{- $schema := (lookup "{$.components.schemas}" $.Document | first | first | values | lookup $definitionPath | first | lookup $definitionPath2 | first | lookup $definitionPath3 | first | first) }}
+
+DESCRIPTION:
+{{ $schema.description | wrap 76 | indent 4}}
+
+FIELDS:{{template "fieldList" (dict "schema" $schema "level" 1 "Document" $.Document "Recursive" $.Recursive)}}
+
+{{- /* Takes dictionary as argument with keys:
+    schema: openapiv3 JSON schema
+    level: indentation level
+    document: entire doc
+    recursive: print recursive
+ */ -}}
+{{- define "fieldList" }}
+    {{- /* Resolve schema if it is a ref */}}
+    {{- $resolved := coalesce (lookup (printf "{$.%v}" (slice (get .schema "$ref" | replace "." "\\." | splitList "/") 1 | joinList ".")) $.Document | first | first) .schema }}
+    {{- range $k, $v := $resolved.properties }}
+        {{- "\n"}}{{indent (int (mul $.level 2)) $k}}{{"\t"}}<{{ coalesce $v.type "Object" }}>
+        {{- if $.Recursive}}
+            {{- template "fieldList" (dict "schema" $v "level" (add $.level 1) "Document" $.Document "Recursive" $.Recursive)}}
+        {{- else}}
+            {{- "\n"}}{{$v.description | wrap (int (sub 80 (mul $.level 4))) | indent (int (mul $.level 4))}}
+        {{- end}}
+    {{- end}}
+    {{- range $resolved.allOf }}
+        {{- template "fieldList" (dict "schema" . "level" $.level "Document" $.Document "Recursive" $.Recursive)}}
+    {{- end}}
+{{- end}}
```

Comment:
I haven't taken the time (and energy) to review the template yet. This will definitely need a test (a bunch of them?). Also make it so that it's easy to test someone's bug report and to create a new test-case from a bug report.